src/hooks/useProductById.ts

```diff
@@ -8,31 +8,42 @@ const useMockData = import.meta.env.USE_MOCK_FOR_API_FAIL === 'true';
 interface UseProductByIdResult {
   product: ProductType | null;
   isLoading: boolean;
+  status?: number; // Optional status code
 }
 
-const fetchProductById = async (productId: string) => {
+const fetchProductById = async (productId: string, useMockData: boolean) => {
   if (!productId) {
-    return null;
+    return { data: null, status: 0 };
   }
 
-  const response = await ProductAPIService.getById(productId);
-
-  if (response.isSuccess && response.data !== undefined && !useMockData) {
-    return response.data;
+  if (useMockData) {
+    const mockProduct = mockProducts.find(product => product.id.toString() === productId) || null;
+    return { data: mockProduct, status: 200 }; // Assuming mock data is always successful
```

Comment:
add break line before return, recheck for all

Reply:
Fixed